src/org/joml/Vector3i.java

```diff
@@ -133,9 +171,9 @@ public Vector3i(Vector2ic v, int z) {
      *          the {@link RoundingMode} to use
      */
     public Vector3i(Vector2fc v, float z, int mode) {
-        x = Math.roundUsing(v.x(), mode);
-        y = Math.roundUsing(v.y(), mode);
-        z = Math.roundUsing(z, mode);
+        this.x = Math.roundUsing(v.x(), mode);
+        this.y = Math.roundUsing(v.y(), mode);
+        this.z = Math.roundUsing(z, mode);
```

Comment:
this resolve a bug with the assignment. this z here shadows the class component. @httpdigest